File: c/Trees/4_levelOrder_.c
#include<stdio.h>
#include<stdlib.h>

struct node {
    int data;
    struct node *left;
    struct node *right;
};
 
/* Function to create a new node */
 
struct node *CreateNode(int data)
{
    struct node *newNode=malloc(sizeof(struct node));
    if (!newNode) {
        return NULL;
    }
    newNode->data = data;
    newNode->left = newNode->right = NULL;
    return newNode;
}
void inorder(struct node *temp)
{
    if (temp == NULL)
        return;
 
    inorder(temp->left);
    printf("%d\t",temp->data);
    inorder(temp->right);
}


void display(struct node *root){
    int node_left=0,node_right=0;
    if(root==NULL){
        return;
    }
    if(root->left !=NULL){
        node_left=root->left->data;
    }
    if(root->right !=NULL){
        node_right=root->right->data;
    }
    printf(" <- %d <-- %d --> %d --> \n",node_left,root->data,node_right);
    display(root->left);
    display(root->right);
}

void levelOrder(struct node *root){
    
}

int main(){
    struct node *root = CreateNode(50);
    root->left = CreateNode(25);
    root->left->left = CreateNode(12);
    root->left->right = CreateNode(37);
    root->left->right->left = CreateNode(30);
    root->right = CreateNode(75);
    root->right->left = CreateNode(62);
    root->right->left->right = CreateNode(70);
    root->right->right = CreateNode(57);

    inorder(root);
    printf("\n");
    display(root);

}

File: Trees/4_levelOrder_.py
from collections import deque

class newNode:
    def __init__(self,data):
        self.data=data
        self.left=None
        self.right=None


def display(head):
    left_node=0.0
    right_node=0.0
    if not head:
        return
    if(head.left):
        left_node=head.left.data
    if(head.right):
        right_node=head.right.data
    print(left_node,"<-",head.data,"->",right_node)
    display(head.right)
    display(head.left)
    
def levelOrder(root):
    q=deque()
    q.append(root)
    while(len(q)>0):
        count=len(q)
        for _ in range(count):
            node=q.popleft()
            print(node.data,end=" ")
            if(node.left !=None):
                q.append(node.left)
            if(node.right !=None):
                q.append(node.right)
        print("\n")
    return

    
    

root = newNode(50)
root.left=newNode(25)
root.right=newNode(75)
root.right.right=newNode(57)
root.right.left=newNode(62)
root.right.left.right=newNode(70)
root.left.right=newNode(37)
root.left.right.left=newNode(37)
root.left.left=newNode(12)
display(root)
levelOrder(root)
